pkg/sbom/io/decode.go

```diff
@@ -340,7 +368,13 @@ func (m *Decoder) addOSPkgs(sbom *types.SBOM) {
 // addLangPkgs traverses relationships and adds language-specific packages
 func (m *Decoder) addLangPkgs(sbom *types.SBOM) {
 	for id, app := range m.apps {
-		app.Packages = append(app.Packages, m.traverseDependencies(id)...)
+		app.Packages = append(m.traverseDependencies(id), app.Packages...)
```

Comment:
The first ones should be packages from components (so that when removing duplicates we take exactly this package), because there should be more information about the component there.